eth/api.go

```diff
@@ -336,6 +338,63 @@ func (api *PrivateDebugAPI) GetBadBlocks(ctx context.Context) ([]*BadBlockArgs,
 	return results, nil
 }
 
+type AccountRangeResult struct {
+	Addresses []common.Address `json:"addresses"`
+	Next      common.Address   `json:"next"`
+}
+
+func accountRange(st state.Trie, start *common.Address, maxResult int) (AccountRangeResult, error) {
+	it := trie.NewIterator(st.NodeIterator(crypto.Keccak256(start[:])))
+	result := AccountRangeResult{Addresses: []common.Address{}, Next: common.Address{}}
+	for i := 0; i < maxResult && it.Next(); i++ {
+		if preimage := st.GetKey(it.Key); preimage != nil {
+			result.Addresses = append(result.Addresses, common.BytesToAddress(preimage))
+		} else {
+			return AccountRangeResult{}, fmt.Errorf("preimage not found for 0x%s", hex.EncodeToString(it.Key))
+		}
+	}
+
+	if it.Next() {
+		if preimage := st.GetKey(it.Key); preimage != nil {
+			result.Next = common.BytesToAddress(preimage)
+		} else {
+			return AccountRangeResult{}, fmt.Errorf("preimage not found for 0x%s", hex.EncodeToString(it.Key))
+		}
+	}
+
+	return result, nil
+}
+
+// enumerate all accounts in the latest state
+func (api *PrivateDebugAPI) AccountRangeAt(ctx context.Context, startAddr *common.Address, maxResults int) (AccountRangeResult, error) {
```

Comment:
Like Martin suggested, please add a constant `MaxResults` and check that `maxResults` is lower than that, otherwise return an error. Yes, the `--debug` flag needs to be enabled, but it doesn't mean that we should let any possible attack vector enabled in case a production server has it enabled for whatever reason... like the ewasm testnet block explorer, for instance.